### ISSUE TITLE
Rest Api gives HTTP 404

### ISSUE THREAD

**Issue by sanuj** [magento/magento2#1474]
I have been trying to test the Rest Api without success. I used RESTClient (Firefox addon) to send a GET request on  `http://localhost/api/rest/products` and received 404 (Not found). My frontend and backend (Admin dashboard) is working perfectly. I am running Magento on LAMP stack on Ubuntu 14.04. Can't find anything anywhere which can solve my problem so I am opening an Issue here.

**Comment by sanuj** [comment]
@vpelipenko any fix for this?

**Comment by kokoc** [comment]
@sanuj Thank you for contacting us. Please try to use path like /index.php/rest/default/V1/products?searchCriteria[page_size]=100

**Comment by sanuj** [comment]
Thanks @kokoc Now i'm getting good xml responses but with internal errors like `Warning: Invalid argument supplied for foreach() in /var/www/magento2/app/code/Magento/Catalog/Model/ProductRepository.php`

I'll try to sort this out. Why am I not able to use the pretty URLs as mentioned in the API documentation?

**Comment by kokoc** [comment]
@sanuj pretty URLs from http://devdocs.magento.com/guides/v1.0/get-started/gs-web-api-request.html works for me. 
Please provide additional information:
* Magento version
* Full error message

**Comment by sanuj** [comment]
@kokoc I cloned magento2 so I guess the version is 2.0. Full error message:
```
[2015-07-15 06:34:57] main.CRITICAL: exception 'Exception' with message 'Report ID: webapi-55a5ff11b9bfc; Message: Warning: Invalid argument supplied for foreach() in /var/www/magento2/app/code/Magento/Catalog/Model/ProductRepository.php on line 671' in /var/www/magento2/lib/internal/Magento/Framework/Webapi/ErrorProcessor.php:194
Stack trace:
#0 /var/www/magento2/lib/internal/Magento/Framework/Webapi/ErrorProcessor.php(139): Magento\Framework\Webapi\ErrorProcessor->_critical(Object(Exception))
#1 /var/www/magento2/app/code/Magento/Webapi/Controller/Rest.php(191): Magento\Framework\Webapi\ErrorProcessor->maskException(Object(Exception))
#2 /var/www/magento2/var/generation/Magento/Webapi/Controller/Rest/Interceptor.php(24): Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http))
#3 /var/www/magento2/lib/internal/Magento/Framework/App/Http.php(115): Magento\Webapi\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#4 /var/www/magento2/lib/internal/Magento/Framework/App/Bootstrap.php(257): Magento\Framework\App\Http->launch()
#5 /var/www/magento2/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#6 {main} [] []
```
I was following this API documentation:
http://www.magentocommerce.com/api/rest/introduction.html

**Comment by sanuj** [comment]
`HTTP Status Code: 500 Internal Server Error`
In response, I get the following xml:
```
<?xml version="1.0"?>
<response>
  <message>Internal Error. Details are available in Magento log file. Report ID: webapi-55a600307f979</message>
</response>
```
Error related to ` Report ID: webapi-55a600307f979` is posted in the above comment.

**Comment by kokoc** [comment]
I can't reproduce this on internal code base, so next code update should fix such errors.
Also you are using Magento 1 documentation, please use document mentioned earlier and `https://github.com/magento/magento2/blob/develop/app/code/Magento/Catalog/etc/webapi.xml` to get available resources.

**Comment by sanuj** [comment]
@kokoc Thanks for your help. I think we should change the documentation as I spent lot of my time trying to make magento1 api work on magento2 :) I got that first link on google when I searched for magento api. Maybe just add a line on the docs specifying the version of magento it is for ;)

**Comment by sanuj** [comment]
I can fix it and send a PR if you can tell me how I can change it.

**Comment by kokoc** [comment]
Magento 1 documentation could be changed only internally. We placed such task in backlog. Internal reference: MTD-2009

**Comment by vladimir-k** [comment]
Hi, @sanuj 
Magento 1 documentation title was changed to "Magento 1.x" and was linked with Magento 2 DevDocs (for ex., [Getting Started with Magento Web APIs](http://devdocs.magento.com/guides/v2.0/get-started/bk-get-started-api.html)).
Thank you for your comments!

**Comment by gg24** [comment]
Use this,

http://localhost/rest/V1/categories/4/products/

where 4 is category Id

**Comment by magento-engcom-team** [comment]
Hi @sanuj. Thank you for your report. 
The issue has been fixed in magento/magento2#28798 by @Nazar65 in 2.4-develop branch
Related commit(s):
 - [a2d51131606d59e3124919f120243cfd69d998f3](https://github.com/magento/magento2/commit/a2d51131606d59e3124919f120243cfd69d998f3)
 - [3bc82395fede9a07a51464ff21c7b467befbdd2e](https://github.com/magento/magento2/commit/3bc82395fede9a07a51464ff21c7b467befbdd2e)
 - [34bff1266d63294251bb96798f2c7cafcaba8122](https://github.com/magento/magento2/commit/34bff1266d63294251bb96798f2c7cafcaba8122)

The fix will be available with the upcoming 2.4.1 release.
